merkle/compact/tree.go

```diff
@@ -26,7 +26,7 @@ import (
 	"github.com/google/trillian/merkle/hashers"
 )
 
-// NodeID identifies a node of a Merkle tree.
+// NodeID identifies a node of a log type Merkle tree.
```

Comment:
What properties of a "log type" Merkle tree does this require to be useful? Is it primarily the density (i.e. not sparse - it doesn't have missing elements)?

Reply:
After some thinking, this `NodeID` type on its own only restricts the max size of the tree (it doesn't have to be log, or dense). I omitted this because it follows from the fact that it's a pair of integers.